worlds/tww/Macros.py

```diff
@@ -0,0 +1,1203 @@
+        and can_defeat_green_chuchus(state, player)
+        and can_defeat_dark_chuchus(state, player)
+    )
+
+
+def can_reach_earth_temple_right_path(state: CollectionState, player: int) -> bool:
+    return (
+        can_reach_earth_temple_sun_statue_room(state, player)
+        and can_play_command_melody(state, player)
+        and state.has("Skull Hammer", player)
+    )
+
+
+def can_reach_earth_temple_left_path(state: CollectionState, player: int) -> bool:
+    return can_reach_earth_temple_sun_statue_room(state, player) and state.has("ET Small Key", player, 2)
+
+
+def can_reach_earth_temple_moblins_and_poes_room(state: CollectionState, player: int) -> bool:
+    return (
+        can_reach_earth_temple_left_path(state, player)
+        and has_fire_arrows(state, player)
+        and state.has("Power Bracelets", player)
+        and can_defeat_floormasters(state, player)
+        and (can_play_command_melody(state, player) or has_mirror_shield(state, player))
+    )
+
+
+def can_reach_earth_temple_basement(state: CollectionState, player: int) -> bool:
+    return (
+        can_reach_earth_temple_sun_statue_room(state, player)
+        and can_play_command_melody(state, player)
+        and can_aim_mirror_shield(state, player)
+    )
+
+
+def can_reach_earth_temple_redead_hub_room(state: CollectionState, player: int) -> bool:
+    return can_reach_earth_temple_basement(state, player) and can_play_earth_gods_lyric(state, player)
+
+
+def can_reach_earth_temple_third_crypt(state: CollectionState, player: int) -> bool:
+    return (
+        can_reach_earth_temple_redead_hub_room(state, player)
+        and (can_play_command_melody(state, player) or can_aim_mirror_shield(state, player))
+        and state.has_all_counts({"Power Bracelets": 1, "Skull Hammer": 1, "ET Small Key": 3}, player)
+        and (can_defeat_red_bubbles(state, player) or state._tww_precise_2(player))
+        and can_play_command_melody(state, player)
+        and can_aim_mirror_shield(state, player)
+    )
+
+
+def can_reach_earth_temple_tall_vine_room(state: CollectionState, player: int) -> bool:
+    return can_reach_earth_temple_third_crypt(state, player) and can_play_earth_gods_lyric(state, player)
+
+
+def can_reach_earth_temple_many_mirrors_room(state: CollectionState, player: int) -> bool:
+    return can_reach_earth_temple_tall_vine_room(state, player)
+
+
+def can_reach_wind_temple_kidnapping_room(state: CollectionState, player: int) -> bool:
+    return (
+        can_play_command_melody(state, player)
+        and state.has("Iron Boots", player)
+        and can_fly_with_deku_leaf_indoors(state, player)
+    )
+
+
+def can_reach_end_of_wind_temple_many_cyclones_room(state: CollectionState, player: int) -> bool:
+    return can_reach_wind_temple_kidnapping_room(state, player) and (
+        (
+            state.has("Iron Boots", player)
+            and can_fan_with_deku_leaf(state, player)
+            and can_fly_with_deku_leaf_indoors(state, player)
```

Comment:
```suggestion
            and can_fly_with_deku_leaf_indoors(state, player)
```
`can_fly_with_deku_leaf_indoors` is exclusively more rules than `can_fly_with_deku_leaf`, so I would recommend just removing the `can_fly_with_deku_leaf` altogether, since it isn't actually relevant

Reply:
This is true, but the the first rule is `can_fan_with_deku_leaf`, not `can_fly_with_deku_leaf`. I would prefer if the rules remained readible by a less-technical audience so that the logic for the location can be more easily interpreted (in this case, I'm assuming it's using the Deku Leaf to stun the Peahats in the room; I haven't done this room vanilla in years). This might be an instance where it makes sense to reduce, however, so I'll consider it, especially if you believe it would be a worthwhile change.

Reply:
Since the idea is to make it more readable by a less technical audience, it's fine then I think. Maybe core will think differently, and maybe it'll end up making it much slower, but whatever lol